File: ecommerce-api-legacy/src/middleware/errorHandler.js
const { AppError } = require('../utils/errors');
const logger = require('../utils/logger');

function errorHandler(err, req, res, next) {
  if (err instanceof AppError) {
    return res.status(err.statusCode).json({ erro: err.message, sucesso: false });
  }

  logger.error('Unhandled exception', {
    path: req.path,
    method: req.method,
    message: err.message,
  });

  return res.status(500).json({ erro: 'Erro interno do servidor', sucesso: false });
}

module.exports = { errorHandler };

File: task-manager-api/middleware/error_handler.py
import logging

from flask import jsonify

from utils.errors import ForbiddenError, NotFoundError, UnauthorizedError, ValidationError


logger = logging.getLogger(__name__)


def register_error_handlers(app):
    @app.errorhandler(ValidationError)
    def handle_validation_error(error):
        return jsonify({"erro": str(error), "sucesso": False}), error.status_code

    @app.errorhandler(NotFoundError)
    def handle_not_found_error(error):
        return jsonify({"erro": str(error), "sucesso": False}), error.status_code

    @app.errorhandler(UnauthorizedError)
    def handle_unauthorized_error(error):
        return jsonify({"erro": str(error), "sucesso": False}), error.status_code

    @app.errorhandler(ForbiddenError)
    def handle_forbidden_error(error):
        return jsonify({"erro": str(error), "sucesso": False}), error.status_code

    @app.errorhandler(ValueError)
    def handle_value_error(error):
        return jsonify({"erro": str(error), "sucesso": False}), 400

    @app.errorhandler(Exception)
    def handle_unexpected_error(error):
        logger.exception("Unhandled exception")
        return jsonify({"erro": "Erro interno do servidor", "sucesso": False}), 500
